code/modules/atmospherics/machinery/components/fusion/hypertorus.dm

```diff
@@ -0,0 +1,1376 @@
+		nodes[2] = null
+		nullifyPipenet(parents[1])
+
+	SetInitDirections()
+	atmosinit()
+	node1 = nodes[1]
+	if(node1)
+		node1.atmosinit()
+		node1.addMember(src)
+	node2 = nodes[2]
+	if(node2)
+		node2.atmosinit()
+		node2.addMember(src)
+	SSair.add_to_rebuild_queue(src)
+	return TRUE
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/proc/check_part_connectivity()
+	. = TRUE
+	if(!anchored || panel_open)
+		return FALSE
+
+	for(var/obj/machinery/hypertorus/object in orange(1,src))
+		if(. == FALSE)
+			break
+
+		if(object.panel_open)
+			. = FALSE
+
+		if(istype(object,/obj/machinery/hypertorus/corner))
+			var/dir = get_dir(src,object)
+			if(dir in GLOB.cardinals)
+				. =  FALSE
+			switch(dir)
+				if(SOUTHEAST)
+					if(object.dir != SOUTH)
+						. = FALSE
+				if(SOUTHWEST)
+					if(object.dir != WEST)
+						. =  FALSE
+				if(NORTHEAST)
+					if(object.dir != EAST)
+						. =  FALSE
+				if(NORTHWEST)
+					if(object.dir != NORTH)
+						. =  FALSE
+			corners |= object
+			continue
+
+		if(get_step(object,turn(object.dir,180)) != loc)
+			. =  FALSE
+
+		if(istype(object,/obj/machinery/hypertorus/interface))
+			if(linked_interface && linked_interface != object)
+				linked_interface = null
+				. =  FALSE
+			linked_interface = object
+
+	for(var/obj/machinery/atmospherics/components/unary/hypertorus/object in orange(1,src))
+		if(. == FALSE)
+			break
+
+		if(object.panel_open)
+			. = FALSE
+
+		if(get_step(object,turn(object.dir,180)) != loc)
+			. =  FALSE
+
+		if(istype(object,/obj/machinery/atmospherics/components/unary/hypertorus/fuel_input))
+			if(linked_input && linked_input != object)
+				linked_input = null
+				. =  FALSE
+			linked_input = object
+
+		if(istype(object,/obj/machinery/atmospherics/components/unary/hypertorus/waste_output))
+			if(linked_output && linked_output != object)
+				linked_output = null
+				. =  FALSE
+			linked_output = object
+
+		if(istype(object,/obj/machinery/atmospherics/components/unary/hypertorus/moderator_input))
+			if(linked_moderator && linked_moderator != object)
+				linked_moderator = null
+				. =  FALSE
+			linked_moderator = object
+
+	if(!linked_interface || !linked_input || !linked_moderator || !linked_output || corners.len != 4)
+		. = FALSE
+
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/proc/activate(mob/living/user)
+	if(active)
+		to_chat(user, "<span class='notice'>You already activated the machine.</span>")
+		return
+	to_chat(user, "<span class='notice'>You link all parts toghether.</span>")
+	active = TRUE
+	update_icon()
+	linked_interface.active = TRUE
+	linked_interface.update_icon()
+	linked_input.active = TRUE
+	linked_input.update_icon()
+	linked_output.active = TRUE
+	linked_output.update_icon()
+	linked_moderator.active = TRUE
+	linked_moderator.update_icon()
+	for(var/obj/machinery/hypertorus/corner/corner in corners)
+		corner.active = TRUE
+		corner.update_icon()
+	soundloop = new(list(src), TRUE)
+	soundloop.volume = 5
```

Comment:
why not have the linked objects check back with the core to see if they should be active? or am I misunderstanding something.

Reply:
i could, but i didn't want them to have processing all the time to listen to the core, is there any other way i could do this?